### PR TITLE
[kjobctl] Add ownerReference for additional Slurm objects.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add ownerReference for additional Slurm objects in kjobctl. This allow to remove additional objects when removing Job.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Follow-up for https://github.com/kubernetes-sigs/kueue/pull/3072#discussion_r1773067656.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/pkg/cmd/create/create.go [hunk 1]
```diff
@@ -590,7 +590,12 @@ func (o *CreateOptions) Run(ctx context.Context, clientGetter util.ClientGetter,
 
 	for i := range objs {
```
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/cmd/create/create.go]: Done

File: cmd/experimental/kjobctl/pkg/builder/builder_test.go [hunk 2]
```diff
@@ -448,7 +448,12 @@ func TestBuilder(t *testing.T) {
 				return
 			}
 
-			if diff := cmp.Diff(tc.wantObj, gotObjs); diff != "" {
+			var gotObjs []runtime.Object
+			if rootObj != nil {
+				gotObjs = append(gotObjs, rootObj)
+			}
+			gotObjs = append(gotObjs, childObjs...)
```
Comment (trasc) [inline cmd/experimental/kjobctl/pkg/builder/builder_test.go:455]: nit:
```suggestion
			gotObjs := append([]runtime.Object{rootObj}, childObjs...)
```

Is there any scenario in which the root object is nil?

(the same goes for some other tests)

**Review by trasc — COMMENTED** [review]
Some nits, otherwise LGTM

File: cmd/experimental/kjobctl/pkg/builder/builder_test.go (hunk 2 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/builder/builder_test.go:455]: Yes, when we testing with errors.

Comment (trasc) [reply cmd/experimental/kjobctl/pkg/builder/builder_test.go:455]: > Yes, when we test it with errors.

... :)

then maybe is better to use `wantRootObject` and `wantChildObjects`

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/builder/builder_test.go]: Done

**Review by trasc — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**